src/test/test_controller.c

```diff
@@ -1689,6 +1691,187 @@ test_download_status_bridge(void *arg)
   return;
 }
 
+/** Mock cached consensus */
+static cached_dir_t *mock_ns_consensus_cache;
+static cached_dir_t *mock_microdesc_consensus_cache;
+
+int
```

Comment:
Why do you have the function prototype and the actual function separate? We don't do that. Please dump the function prototype and just document the function directly. Same for the ones below.

Reply:
These are not functions, these are variables that I create outside the test case so that I can free them. The original code:

```
  if (we_want_to_fetch_flavor(get_options(), flavor)) {
    /** Check from the cache */
    const cached_dir_t *consensus = dirserv_get_consensus(flavor_name);
    if (consensus) {
      *answer = tor_strdup(consensus->dir);
    }
  }
```
... doesn't free the consensus as it points to a global object. I keep these mock variables for the sake of avoiding memory leaks from within the test by freeing them after use.